Include `ERC20Permit` in `SolidStateERC20`

ItsNickBarry opened the pull request:
Originally described in #105, but no longer in scope.  Merge after #118.

0xCourtney commented:
IIUC, the standard is to provide each internal contract with an internal interface, even if it's empty?

ItsNickBarry commented:
@0xCourtney Maybe.  I added the empty files to make it easier to methodically check for and apply inheritance ordering, but I removed some and compilation still worked.
